lib/spack/env/cc

```diff
@@ -162,7 +162,7 @@ fi
 # It doesn't work with -rpath.
 # This variable controls whether they are added.
 add_rpaths=true
-if [[ mode == ld && $OSTYPE == darwin* ]]; then
+if [[ $mode == ld && "$SPACK_SHORT_SPEC" =~ "darwin" ]]; then
```

Comment:
@eschnett: this was broken by your changes to `cc` since f80e839ff (when I last fixed the `ld -r` thing for @gartung).  If you look at f80e839ff, the OSTYPE check is higher up in the file and is not `&&`'d with the mode check.  Do a `git diff f80e839ff cc` to see.

I should've tested the `gcc` build -- luckily they are imaging me a mac to do automated testing on at LLNL so I can hopefully get some better overnight tests going.  I don't think I can build things as big as `gcc` on Travis, can I?  I think it'll exceed the max build time.  Still, we could probably add a regression test for this behavior in `cc`.

If the new syntax is wrong, can you submit a fix and check that it builds gcc 4.9.3 on your machine?